### PR TITLE
Enhance docs and add a few methods for `option.sw`

### PR DESCRIPTION
**Pull request by mohammadfawaz**
A lot of this is copied from https://doc.rust-lang.org/src/core/option.rs.html#1864

Now that we have `forc-doc`, we want to spend some time enhancing all of our standard library docstrings.

There are a few more methods we can implement. I attempted `transpose()` but faced an [issue](https://github.com/FuelLabs/sway/issues/3321). There are all the Boolean methods such as `and` and `or` as well.

- [x] Write tests for `unwrap_or` and `ok_or`.
- [x] Enhance tests for `Option` in general

### CODE REVIEW

File: sway-lib-std/src/option.sw [hunk 1]
```diff
@@ -1,53 +1,221 @@
...
     /////////////////////////////////////////////////////////////////////////
-    /// Returns `true` if the result is [`Some`].
-    fn is_some(self) -> bool {
+
+    /// Returns `true` if the option is a [`Some`] value.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// let x: Option<u32> = Some(2);
```
Comment (SwayStar123) [inline sway-lib-std/src/option.sw:105]: I dont think Sway has the variants of Option in its prelude, so `Option::Some(2)` would be more accurate

Comment (mohammadfawaz) [reply sway-lib-std/src/option.sw:105]: Yup you're right.. I was fixing those but clearly forgot this one. In fact, we can't import enum variants yet either which is unfortunate, but that's coming!

File: sway-lib-std/src/option.sw [hunk 2]
```diff
@@ -1,18 +1,86 @@
-//! Error handling with the `Option` type.
+//! Optional values.
 //!
-//! [`Option<T>`][`Option`] is the type used for representing the existence or absence of a value. It is an enum with the variants, [`Some(T)`], representing
-//! some value, and [`None()`], representing
-//! no value.
+//! Type [`Option`] represents an optional value: every [`Option`]
+//! is either [`Some`] and contains a value, or [`None`], and
+//! does not. [`Option`] types are very common in Rust code, as
```
Comment (AlicanC) [inline sway-lib-std/src/option.sw:5]: Should this say "Rust" or "Sway"?

**Review by AlicanC — APPROVED** [review]

File: sway-lib-std/src/option.sw (hunk 2 above)
Comment (mohammadfawaz) [reply sway-lib-std/src/option.sw]: Ah of course. Updated.

File: sway-lib-std/src/option.sw [hunk 3]
```diff
@@ -1,18 +1,86 @@
...
 //!
-//! [`Option<T>`][`Option`] is the type used for representing the existence or absence of a value. It is an enum with the variants, [`Some(T)`], representing
-//! some value, and [`None()`], representing
-//! no value.
+//! Type [`Option`] represents an optional value: every [`Option`]
+//! is either [`Some`] and contains a value, or [`None`], and
+//! does not. [`Option`] types are very common in Sway code, as
+//! they have a number of uses:
+//!
+//! * Initial values
```
Comment (emilyaherbert) [inline sway-lib-std/src/option.sw:8]: Maybe expand this to "Initial values can be set to [`None`]"  or something like that.

File: sway-lib-std/src/option.sw [hunk 4]
```diff
@@ -1,18 +1,86 @@
...
+//! ```
+//! fn divide(numerator: u64, denominator: u64) -> Option<u64> {
+//!     if denominator == 0 {
+//!         Option::None
+//!     } else {
+//!         Option::Some(numerator / denominator)
+//!     }
+//! }
+//!
+//! // The return value of the function is an option
```
Comment (emilyaherbert) [inline sway-lib-std/src/option.sw:26]: Maybe put the remaining block of code in its own function, like `call_divide` or something like that?

**Review by emilyaherbert — DISMISSED** [review]
Very well written!

**Review by emilyaherbert — APPROVED** [review]

**Review by otrho — APPROVED** [review]

**Pull request merged**